Authentication/AuthenticationCoordinator.swift

```diff
@@ -105,13 +105,13 @@ public extension AuthenticationCoordinator {
     internal func createLoginControllerConfiguration() -> LoginControllerConfiguration {
         let loginViewConfiguration = _componentsFactory.createLoginViewConfiguration()
         let loginViewDelegate = _componentsFactory.createLoginViewDelegate(withConfiguration: loginViewConfiguration)
-        let createLoginView: () -> LoginViewType = { [unowned self] in self._componentsFactory.createLoginView(withDelegate: loginViewDelegate) }
         return LoginControllerConfiguration(
-            viewModel: _componentsFactory.createLoginViewModel(withSessionService: sessionService,
-                credentialsValidator: _componentsFactory.createLogInCredentialsValidator()),
-            viewFactory: createLoginView,
+            viewModelFactory: _componentsFactory.createLoginViewModelFactory(withSessionService: sessionService,
+                                                                             credentialsValidator: _componentsFactory.createLogInCredentialsValidator()),
+            viewFactory: _componentsFactory.createLoginViewFactory(withDelegate: loginViewDelegate),
             transitionDelegate: _componentsFactory.createLoginControllerTransitionDelegate() ?? self,
-            delegate: _componentsFactory.createLoginControllerDelegate())
+            delegate: _componentsFactory.createLoginControllerDelegate(),
+            loginProviders: _componentsFactory.createLoginProviders())
```

Comment:
If the providers are created here, why can't the view model be created directly with them instead of passing them to the configuration?